### PR TITLE
28 add tests for check hoksso

### PR DESCRIPTION
**Pull request by philsmart**

### CODE REVIEW

File: tests/xml/hoksso/good_sp_hoksso.xml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        <AssertionConsumerService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Artifact" Location="https://sp.example.org/Shibboleth.sso/SAML2/Artifact" index="2"/>
+        <AssertionConsumerService Binding="urn:oasis:names:tc:SAML:2.0:bindings:PAOS" Location="https://sp.example.org/Shibboleth.sso/SAML2/ECP" index="3"/>
+        <AssertionConsumerService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="https://sp.example.org/Shibboleth.sso/SAML2/POST" index="4"/>
+        <AssertionConsumerService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST-SimpleSign" Location="https://sp.example.org/Shibboleth.sso/SAML2/POST-SimpleSign" index="5"/>  
+
+        <!-- HoK elements -->
+        <AssertionConsumerService xmlns:hoksso="urn:oasis:names:tc:SAML:2.0:profiles:holder-of-key:SSO:browser" 
+            Binding="urn:oasis:names:tc:SAML:2.0:profiles:holder-of-key:SSO:browser" 
+            Location="https://sp.example.org/Shibboleth.sso/HoKSSO" 
+            hoksso:ProtocolBinding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Artifact" index="5"/>
```
[the remark this answers is not in the document]
Comment (philsmart) [reply tests/xml/hoksso/good_sp_hoksso.xml:98]: I will fix that for these tests, as we will need specific ones when that checking ruleset is enabled.

[the remark this answers is not in the document]
Comment (philsmart) [reply tests/xml/hoksso/good_sp_hoksso.xml:98]: Ok, I fixed that up

**Review by iay — CHANGES_REQUESTED** [review]
We talked back in August on a previous PR about non-terminated XML files and Git's slight disdain for them. I note that this has happened again this time round.

Back then you said something about not being able to sync VS Code settings across machines, so I'm assuming that's the issue here again. Can we fix these files for this commit?

Happy to accept this PR with that change (or without, actually, it's not really a big deal).

**Comment by philsmart** [comment]
Added a final new line at the end of the file.

**Review by iay — APPROVED** [review]

**Pull request merged**